### DISCUSSION TITLE
🚨 Request for Ideas & Questions for Inspire 2024 Developer Round Tables.

### DISCUSSION THREAD

**Discussion by drietsch** [pimcore/pimcore#16559] [General]
https://github.com/pimcore/pimcore/assets/62896/34e4889e-0c5d-481e-b413-fe8638e54411

```
  _____ _                                _____                 _          
 |  __ (_)                              |_   _|               (_)         
 | |__) | _ __ ___   ___ ___  _ __ ___    | |  _ __  ___ _ __  _ _ __ ___ 
 |  ___/ | '_ ` _ \ / __/ _ \| '__/ _ \   | | | '_ \/ __| '_ \| | '__/ _ \
 | |   | | | | | | | (_| (_) | | |  __/  _| |_| | | \__ \ |_) | | | |  __/
 |_|   |_|_| |_| |_|\___\___/|_|  \___| |_____|_| |_|___/ .__/|_|_|  \___|
                                                        | |               
                                                        |_|               
``` 
Hi guys! This time commander Dietz, CEO of Pimcore, is in the house. You probably already hopefully know that the **[Pimcore Inspire Developer Conference 2024](https://pimcore.com/en/inspire-2024)** is taking place in [Zell am See, Austria](https://pimcore.com/en/inspire-2024/location) from March 19-20, 2024. Everybody, that has been to the Inspire before, knows that this is quite a crazy event, powered by a lot of Pimcore passion. It's the same this year - promised. Additionally we announce this year quite some awesome stuff, such as **Pimcore Studio**, the new UX/UI of the management interface, the **Pimcore Copilot**, quite some AI/ML and workflow automation stuff and a lot more. Just check out the [full agenda](https://pimcore.com/en/inspire-2024/agenda-2024)! It's gonna be great and I'm really looking forward meeting you guys at the Alpine Expedition, which is the motto of this event.

🚨 Anyhow, I need your help and your ideas. This years' Inspire conference has a strong focus on development and technology. We will have development round tables on the second day. This round tables will be the heart of the development community and we imagine an open dialog with the whole development community. Therefore this part of the agenda of the conference is actually defined by all of us. **Therefore I would like to invite you, to post your questions and ideas 💬 within this thread.** If you have a special topic and would also like to have your own round table, just drop me a line at [dietmar.rietsch@pimcore.com](mailto:dietmar.rietsch@pimcore.com). Anyhow: I'll bring a keg of beer to the Inspire, and the best ideas are rewarded 🏆 by me - if you dare ;-). So let's kick it off and post your questions and ideas, that you want to have discussed on the Inspire. Thanks a lot! Cheers, Dietz 🚀💜

`Now let us hear your input for the Inspire 2024 Developer Round Tables!`

**Comment by dpfaffenbauer** [comment]
```
                                      %%%%%%%%%%                                    
                                   %%%%%%%%%%%%%%%%%%%                               
                               %%%%%%%%%%%%%%%%%%%%%%%%%%%                           
                           %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                       
                      %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                   
                  %%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%              
             %%%%%%%%%%%%%%%%%%%%%%         %        %%%%%%%%%%%%%%%%%%%%%%          
         %%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%      
     %%%%%%%%%%%%%%%%%%%%%          %%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%% 
    %%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%%%%%%        %%%%%%%%%%%%%%%%%% 
    %%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%% 
    %%%%%%%%%%%        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%% 
    %%%%%%%%%%%      %%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%        %%%%% 
    %%%%%%%%%%%      %%%%%%%%%%%%%%%                 %%%%%%%%%%%%%%%%%%%%%%        % 
    %%%%%%%%%%%      %%%%%%%%%%%                         %%%%%%%%%%%%%%%%%%%%%%      
    %%%%%%%%%%%      %%%%%%%%%%                               %%%%%%%%%%%%%%%%%%%%%  
    %%%%%%%%%%%      %%%%%%%%%%                                   %%%%%%%%%%%%%%%%%% 
    %%%%%%%%%%%        %%%%%%%%                          %%%%         %%%%%%%%%%%%%% 
    %%%%%%%%%%%%%%%         %%%                          %%%%%%%%%        %%%%%%%%%% 
    %%%%%%%%%%%%%%%%%%%                                  %%%%%%%%%%%      %%%%%%%%%% 
      %%%%%%%%%%%%%%%%%%%%%%                             %%%%%%%%%%%      %%%%%%%%%% 
          %%%%%%%%%%%%%%%%%%%%%%                         %%%%%%%%%%%      %%%%%%%%%% 
    %%        %%%%%%%%%%%%%%%%%%%%%%                %%%%%%%%%%%%%%%%      %%%%%%%%%% 
    %%%%%%         %%%%%%%%%%%%%%%%%%%%%        %%%%%%%%%%%%%%%%%%%%      %%%%%%%%%% 
    %%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%        %%%%%%%%%% 
    %%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%% 
    %%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%% 
     %%%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%          %%%%%%%%%%%%%%%%%%%%%  
          %%%%%%%%%%%%%%%%%%%%%%        %%%%%%%%         %%%%%%%%%%%%%%%%%%%%%%      
              %%%%%%%%%%%%%%%%%%%%%%                 %%%%%%%%%%%%%%%%%%%%%%          
                   %%%%%%%%%%%%%%%%%%%%%         %%%%%%%%%%%%%%%%%%%%%               
                       %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                   
                           %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                       
                               %%%%%%%%%%%%%%%%%%%%%%%%%%                            
                                    %%%%%%%%%%%%%%%%%                                
                                        %%%%%%%%%                                                        
```

 - eCommerce with Pimcore ;)
 - Testing with Pimcore (Unit, Behat, etc.)
 - Pimcore at Scale
 - Bundle Development best practices: although, might change a bit depending on Pimcore Studio
 - Community Bundles in General
 - Pimcore do's and dont's for Developers
 - Pimcore <3 Containers

Just some ideas ;)

**Reply by drietsch** [reply]
Power it up, @dpfaffenbauer. That's truly some good stuff. Thanks. I'm just curious, what would be your single most favorite topic, except for CoreShop of course? ;) Have you checked out the detailed agenda yet? What would you think about a special developer lunch action on the second day?

**Reply by dpfaffenbauer** [reply]
To be real here: As a developer, most sessions except the one where you talk about the new Pimcore Studio sound boring or it is a Marketing Channel for a Sponsor (don't take this wring, its not bad, but boring for devs). Laracon fits 11 sessions into a day. You have 2 on day 1 and 4 on day 2.

Developers want insights, check out Symfonycon or Laracon. Every Session is about half an hour from a different community member. There is so much you can take home and use on a daily basis. So far that I would even pay for a remote ticket and watch the sessions. Whereas for Pimcore Inspire, it was always very Marketing intensive, so I never really watched most talks but just hanged out with other Partners and "networked".

I would have loved a more in-depth developer experience approach. More practical things for devs. Things like FrankenPHP with Pimcore, Kubernetes, Bundles like obviously CoreShop, Dachcom Bundles like Toolbox, DynamicSearch, etc. Learnings from Pimcore at scale or How you should test your bundle. 

Pimcore has a very small but still powerful Community with lots of knowledge and insights. Give them a stage to show what they offer and how it works in real life. There is so much everybody in the community can learn from each other, but so little opportunity to do so. 

Maybe a separate Pimcorecon would help there ;)

For the Dev Round Table you only allocated 2 hours, so not much. Hard to say what my absolute favourite is. Lunch should be about lunch ;). No dev is able to process any thing with an empty stomach :P And usually people talk and want to enjoy there food, not sure if can actually something into a lunch break.

My single most favourite: If I could decide what we would talk about for an hour long: I would invite the backbone of the Open Source Community: Bundle maintainers and Core Contributors. Let the Bundle maintainers show what they did, how they did it and how it works in practice. Not as advertisment for the agency, but as actual use cases for good Extensions. Open Source and Free Extensions only of course. Core Contributors can tell their story to how and why they invest their free time or work-time to contribute for "free".

Second most favorite: Kubernetes and Pimcore at Scale ;)

**Reply by dpfaffenbauer** [reply]
PS: behind every pull request is a story that is crazy enough to be told ;)

**Reply by drietsch** [reply]
> PS: behind every pull request is a story that is crazy enough to be told ;)

That surely is that way. You should have a look at my private playground repos. Crazy stories ;-)

Thank you very much for your detailed response and also your constructive criticism. That's what these round tables should also be about. Open communication how things can be improved. Quite simple as that.

@dpfaffenbauer, you have been participating almost all official Pimcore events. And you know where Inspire is coming from. Inspire has always been a partner-centric conference with a strong legacy of business and marketing connotation and with a bit of future-looking bullshit from my end. Being brutally honest here ;-). Do we believe that Pimcore Inspire can greatly improve in terms of developer experience? Surely, it can. And we will take the first step with this years' conference. Simply learn and improve. Based on the valuable input from the community. 

At least for me, comparing Inspire to Symfonycon/Laracon is a double-edged sword. Symfony and Laravel are foundational PHP frameworks, while Pimcore is an application (incl. framework ;-) for data and experience management. But don't get me wrong, I also exactly got your point. Let's bring an idea like "Pimcorecon" to the table then.

I have just rechecked the agenda again and I think we could think about adding another development-centric track to the agenda from 13:00 to 14:30 on the second day. But let's wait for the additional input from the community and then decide.

Thanks again, Dominik, for your honest feedback. Highly appreciated. 🚀

**Comment by andreas-gruenwald** [comment]
We'd love to hear something about "Pimcore + composable commerce".
And in general, how a real-world microservice approach in Pimcore would look like, including dxp and personalization components (storyblok etc.). How would the admin backend in such an approach look like, and how could that work out with Pimcore Studio and the Copilot?

Other ideas:
 - Hugging face use cases and scenarios for our clients
 - Platform agnostic hosting / platform.sh
 - Migration path Pimcore 11 <-> Pimcore Data Studio

**Reply by dpfaffenbauer** [reply]
What is the point of using Pimcore with DXP and personalization together with storyblok?

**Reply by andreas-gruenwald** [reply]
That would be a perfect opening question for the round table discussion.  :-) 

Imho Pimcore has some limitations regarding personalization (search, recommendations, content personalisation, etc.). Connecting external personalization components / APIs (best of breed) could help to improve personalization experience. But if you connect external systems, how does that play together with content, and which system would then be best suited to manage and deliver (headless) personalized content (Pimcore? Can external systems such as storyblok add value, as they already provide headless content APIs? Is there actually any value in adding such systems? ...).

**Reply by dpfaffenbauer** [reply]
@andreas-gruenwald Got your point, good idea

**Reply by drietsch** [reply]
Thanks!!
